static/js/rendered_markdown.js

```diff
@@ -153,6 +153,19 @@ exports.update_elements = (content) => {
         }
     });
 
+    content.find('div.spoiler-header').each(function () {
+        // If a spoiler block has no header content, it should have a default header
+        // We do this client side to allow for i18n by the client
+        if ($.trim($(this).html()).length === 0) {
+            $(this).append(i18n.t('Spoiler'));
+        }
+
+        // Add the expand/collapse button to spoiler blocks
+        const toggle_button_html = '<a class="spoiler-button" aria-expanded="false"><span class="spoiler-arrow spoiler-button-closed"></span></a>';
```

Comment:
Does it just make sense to use a template here to build up the html that goes inside of `<div class="spoiler-header">whatever server sends</div>`?  And then we can replace with a single call to `$('.spoiler-header').html(html)`?

And the server could possibly put some fallback value in the div like `_("Spoilers not supported on this client")`.

Reply:
(the reason I make the above suggestion is that I am always a little worried about append/prepend causing tiny regressions)

Reply:
If the message sender specifies header text then the header stored on the server won't be empty and the contents will matter.

We could still replace it without using `prepend` in some ways; for instance, we could have the server add in a `<div class="button-placeholder">` or something similar.

What are your concerns with `append`/`prepend`?

Reply:
To be clear, I don't have any show-stopping concerns here.  I just feel like sometimes `append` can be slightly harder to understand and/or easier to break.

Reply:
I think the template ends up being messy, and this logic does do precisely what we want.  I do wish we could just use `.text()` to replace the text in a paragraph.  Would it make sense to have an empty `<p></p>` tag be the base header content in our markdown system, rather than having nothing inside?  That way, this logic could expect that `p` tag to be there, and just do `$(".spoiler-header p").text(i18n.t("Spoiler"))`.  

(We generally try to avoid using `.html()` where we can, because it's a place you have to think about malicious HTML.